.github/workflows/pull-requests.yml

```diff
@@ -590,23 +589,144 @@ jobs:
+                echo "DAG L1 nodes are not health, check the logs"
+                exit 1
+              fi
+              echo "DAG L1 nodes not health yet, trying again in 10s:"
+              sleep 10
+            fi
+          done 
+
+      - name: Check Currency L0 nodes health
+        run: |
+          for ((i = 1; i <= 12; i++)); do
+            if curl -s http://127.0.0.1:9400/node/info | grep -o "\"state\":\"Ready\"" >> /dev/null && curl -s http://127.0.0.1:9500/node/info | grep -o "\"state\":\"Ready\"" >> /dev/null && curl -s http://127.0.0.1:9600/node/info | grep -o "\"state\":\"Ready\"" >> /dev/null; then
+              echo "Currency L0 nodes health and Ready"
+              exit 0 
+            else
+              if [ $i -eq 12 ]; then
+                echo "Currency L0 nodes are not health, check the logs"
+                exit 1
+              fi
+                echo "Currency L0 nodes not health yet, trying again in 10s:"
+                sleep 10
+            fi
+          done 
+
+      - name: Check Currency L1 nodes health
+        run: |
+          for ((i = 1; i <= 12; i++)); do
+            if curl -s http://127.0.0.1:9700/node/info | grep -o "\"state\":\"Ready\"" >> /dev/null && curl -s http://127.0.0.1:9800/node/info | grep -o "\"state\":\"Ready\"" >> /dev/null && curl -s http://127.0.0.1:9900/node/info | grep -o "\"state\":\"Ready\"" >> /dev/null; then
+              echo "Currency L1 nodes health and Ready"
+              exit 0 
+            else
+              if [ $i -eq 12 ]; then
+                echo "Currency L1 nodes are not health, check the logs"
+                exit 1
+              fi
+                echo "Currency L1 nodes not health yet, trying again in 10s:"
+                sleep 10
+            fi
+          done       
+
+      - name: Check Currency L0 nodes ordinals
+        run: |
+          export CL0_1_ORDINAL=$(curl -s http://127.0.0.1:9400/snapshots/latest -H "Accept: application/json" | grep -o -m 1 "\"ordinal\":[0-9]*" | grep -o -E "[0-9]+")
+          export CL0_2_ORDINAL=$(curl -s http://127.0.0.1:9500/snapshots/latest -H "Accept: application/json" | grep -o -m 1 "\"ordinal\":[0-9]*" | grep -o -E "[0-9]+")
+          export CL0_3_ORDINAL=$(curl -s http://127.0.0.1:9600/snapshots/latest -H "Accept: application/json" | grep -o -m 1 "\"ordinal\":[0-9]*" | grep -o -E "[0-9]+")
+
+          export SMALLEST_ORDINAL=$CL0_1_ORDINAL
+          if [[ $CL0_2_ORDINAL < $SMALLEST_ORDINAL ]]; then
+            export SMALLEST_ORDINAL=$CL0_2_ORDINAL
+          fi
+          if [[ $CL0_3_ORDINAL < $SMALLEST_ORDINAL ]]; then
+            export SMALLEST_ORDINAL=$CL0_3_ORDINAL
+          fi
+          
+          export BIGGEST_ORDINAL=$CL0_1_ORDINAL
+          if [[ $CL0_2_ORDINAL > $BIGGEST_ORDINAL ]]; then
+            export BIGGEST_ORDINAL=$CL0_2_ORDINAL
+          fi
+          if [[ $CL0_3_ORDINAL > $BIGGEST_ORDINAL ]]; then
+            export BIGGEST_ORDINAL=$CL0_3_ORDINAL
+          fi
```

Comment:
Maybe you could do something like creating an array out of all the ordinals, sorting it numerically and taking first and last as min and max, to not have multiple if statements?